### PR TITLE
Use redux

### PR DESCRIPTION
**Pull request by Umuhan**
### Below are the changes made to the code:
- [x] Wrapped App with the Provider component from react-redux
- [x] Passed  configured store into the Provider
- [x] Created an initialState variable for  booksSlice, which will include the following eg:
-  // Initial state:
  [
    {
    "item_id": "item1",
     "title": "The Great Gatsby",
     "author": "John Smith",
    "category": "Fiction"
   },
 
 ]
- [x] Displayed books, received from the slice, in a reusable component
- [x] Dispatched actions using useDispatch
- [x] Added a Button component

### CODE REVIEW

File: src/components/NewBooks.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  const handleAddBook = () => {
+    dispatch(addBook(newBook));
+    setNewBook({
+      id: '',
+      title: '',
+      author: '',
+      category: '',
+    });
+  };
+
```
Comment (thecodechaser) [inline src/components/NewBooks.js]: - [ ] I'm not able to add a book, It's giving the below error on the console. Please make sure to fix this issue and users should be able to add books without any error.

image

File: src/components/NewBooks.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        <button type="button" onClick={handleAddBook} className="btn-add btn">
+          Add Book
+        </button>
+        <button
+          type="button"
+          className="btn-remove btn"
+          onClick={() => handleRemoveBook(newBook.item_id)}
+        >
+          Remove Book
+        </button>
```
Comment (thecodechaser) [inline src/components/NewBooks.js]: - [ ] You should not show the remove button here, You should handle the remove book on the book's remove button itself that is present in the `Book` component. Please make sure to fix this issue and users should be able to delete books by clicking on the remove button in the `Book` component.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Umuhan👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with Redux code. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/NewBooks.js (hunk 1 above)
Comment (Umuhan) [reply src/components/NewBooks.js]: Thanks, solved

File: src/components/NewBooks.js (hunk 2 above)
Comment (Umuhan) [reply src/components/NewBooks.js]: Thanks, Solved

**Review by shahadat3669 — APPROVED** [review]
Hi @Umuhan, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ Redux `store` has been passed into the `<Provider>`
✅ an `initialState` has been added to the `booksSlice`
✅ Work on the previous reviewer request
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**